.github/workflows/default.yaml

```
@@ -109,10 +109,40 @@ jobs:
           sudo apt-get --quiet=2 build-dep squid
           sudo apt-get --quiet=2 install linuxdoc-tools libtool-bin
 
+      - name: Install prerequisite MacOS packages
+        if: runner.os == 'macOS'
+        run: |
+          brew install \
+            automake coreutils cppunit gawk \
+            gnu-getopt gnu-sed grep libtool \
+            make openldap openssl cyrus-sasl
+
       - name: Checkout sources
         uses: actions/checkout@v4
 
-      - run: ./test-builds.sh
+      - name: Run build on Linux
+        if: runner.os == 'Linux'
+        run: ./test-builds.sh
+
+      - name: Run build on MacOS
+        if: runner.os == 'macOS'
+        run: |
+          eval `brew shellenv`
+          PKG_CONFIG_PATH="$HOMEBREW_PREFIX/lib/pkgconfig"
+          PKG_CONFIG_PATH="$PKG_CONFIG_PATH:$HOMEBREW_PREFIX/opt/openldap/lib/pkgconfig"
+          PKG_CONFIG_PATH="$PKG_CONFIG_PATH:$HOMEBREW_PREFIX/opt/cyrus-sasl/lib/pkgconfig"
+          export PKG_CONFIG_PATH
+          export GETOPT="$HOMEBREW_PREFIX/opt/gnu-getopt/bin/getopt"
+          if gcc --version | grep -q clang ; then
```

Is it possible to test `c++` here instead of `gcc`? It would make this hack less odd/controversial and more readable:

```suggestion
          if c++ --version | grep -q clang ; then
```

P.S. I am surprised this is how we are supposed to detect `-Wno-compound-token-split-by-macro` option support, but I am not going to fight for better detection in this rather constraint environment, despite the fact that, in a constraint environment like this, we do not need to be testing which compiler we were given -- we should know/assume the right answer!

---

We can test c++, but it's really a C problem.
The environment we are targeting is:
```
muno master/src ‹master› » gcc --version
Apple clang version 15.0.0 (clang-1500.3.9.4)
Target: arm64-apple-darwin23.4.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

Using c++ has no real advantage:
```
c++ --version
Apple clang version 15.0.0 (clang-1500.3.9.4)
Target: arm64-apple-darwin23.4.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

If anything, we might want to try "cc" out instead of "gcc", but then Xcode installs a gcc-compatible binary.


> we do not need to be testing which compiler we were given -- we should know/assume the right answer
Sure, we can unconditionally define the flag. The point is: without it, libltdl fails the build

---

Using correct vendor-neutral/non-gcc compiler name would address this change request. If that name should be cc rather than c++, please use cc, of course.

---

And using correct unconditional flags is also fine.